### PR TITLE
Adapt "susemanager/" directory to be Python 2/3 compatible

### PR DESCRIPTION
**Pull request by meaksh**
## What does this PR change?

This PR adapts the code under `susemanager/` directory to be Python 2/3 compatible. Most of the Python code here is deployed by the `susemanager-tools` package which provides the `mgr-sync`, `mgr-create-bootstrap-repo`, `mgr-delete-patch`, `mgr-delete-old-patches` and `mgr-register` scripts.

## Test coverage
- No tests: *No feature added, only porting code**
- Unit tests were added
- Cucumber tests were added

- [X] **DONE**

## Links

Tracks https://github.com/SUSE/spacewalk/issues/5518

- [X] **DONE**

### CODE REVIEW

**Comment by mcalmer** [comment]
@meaksh after a quick look I see one problem.

Spacewalk-backend-libs is a clienttools package and need to be available in python2 and python3.
The current change build either python2 or python3. I think we need to build both at  the same point in time.

**Comment by meaksh** [comment]
@mcalmer - Thanks for clarifying :+1: I wasn't aware this `spacewalk-backend-libs` package is a client tool package which needs to be available on the clients.

I've pushed some new changes which make "spacewalk-backend-libs" client tool package back to be built for both Python 2 and 3 on `suse_version > 1320`:

- `spacewalk-backend-libs` -> Python 2 package
- `python3-spacewalk-backend-libs` -> Python 3 package

Also `susemanager_unittests` are now green.

I think there is nothing else missing here, could you please take a look when have some time? Thanks!

**Comment by mcalmer** [comment]
@meaksh - I think we should merge it as soon as possible. But the cucumber tests should run. So we need the new python-python-pam for SLE12 as well to make it build there.
Currently SLE12 is the only working version.

If this works, we can merge after the last small issues are fixed.

**Comment by mcalmer** [comment]
@meaksh - maybe you can ask the Retail Team if they still need Devel:Galaxy:Manager:TEST. If not, we could try to setup this branch there and run the cucumber tests before they are merged.

File: backend/spacewalk-backend.spec [hunk 1]
```diff
@@ -414,37 +487,50 @@ Group:          Applications/Internet
...
 Requires:       python2-spacewalk-usix
+%endif
 Requires:       subscription-manager
 
 %description cdn
 Tools for syncing content from Red Hat CDN
 
 %prep
-%setup -q
+%setup -q -n spacewalk-backend-4.0.1devel
```
[the remark this answers is not in the document]
Comment (meaksh) [reply backend/spacewalk-backend.spec]: Ooops! Sure, this a mistake.

**Comment by mcalmer** [comment]
@meaksh - main work was on the susemanager package. So a changelog for that with port to python3 would be good.

**Comment by meaksh** [comment]
@mcalmer - a changelog entry for the `susemanager` package is already included here: https://github.com/uyuni-project/uyuni/pull/27/files#diff-91f01447316bf39dcf815c50809c2035R1

Do we need some other adjustment here?

Btw, I just checked with Jiri and it seems they're not longer using "Devel:Galaxy:Manager:TEST" but I'm waiting a final confirmation from Vladimir.

**Pull request merged**